### PR TITLE
don't leak goroutines on redis closed connections

### PR DESCRIPTION
**Pull request by steved**
Even if the connection is closed, there's still a pool in redis.v3.

### CODE REVIEW

File: buckets/redis/bucket.go [hunk 1]
```diff
@@ -114,11 +114,12 @@ func (b *redisBucket) Take(requested int64, maxWaitTime time.Duration) (time.Dur
 			waitTime = time.Nanosecond * time.Duration(waitTimeNanos)
 			keepTrying = false
 		default:
+			// Always close connections on errors to prevent results leaking.
+			b.factory.client.Close()
```
Comment (maniksurtani) [inline buckets/redis/bucket.go]: Can this error? Do we care?

Comment (steved) [inline buckets/redis/bucket.go]: It can, but I don't know if there's anything we can do about it. I can add a log statement.

Comment (maniksurtani) [inline buckets/redis/bucket.go]: Yeah we should at least log it.

Comment (steved) [inline buckets/redis/bucket.go]: Added

**Pull request merged**